app/services/check_assignment_status.rb

```diff
@@ -1,88 +1,126 @@
 # frozen_string_literal: true
-
 # Checks whether expected sandboxes have been created,
 # or if they've been created already, checks their
 # current size and location
 class CheckAssignmentStatus
   def self.check_current_assignments
     return unless Features.wiki_ed?
 
-    Course.current.each do |course|
-      course.assignments.where.not(user: nil).each do |assignment|
-        new(assignment)
+    assignments = Assignment.joins(:course).merge(Course.current).where.not(user: nil)
+    sandboxes_to_check = collect_sandboxes(assignments)
+
+    process_sandboxes(sandboxes_to_check)
+  end
+
+  class << self
```

Comment:
I'd prefer not to use `class << self` here, as the extra nesting makes the code a little harder to read (since we don't use this format very many other places in the codebase).

Reply:
I think the code would be a little simpler if instances were still used, but instead of going one assignment at a time, it went one batch at a time, so that the each instance is expected to make one API request and the page_infos for that request can be an instance variable.